src/coreComponents/fileIO/timeHistory/PackCollection.cpp

```diff
@@ -158,28 +186,28 @@ void PackCollection::updateSetsIndices( DomainPartition & domain )
       {
         for( localIndex idx = 0; idx < ownedIndices.size(); ++idx )
         {
-          if( m_setsIndices[ setIdx ][ idx ] != ownedIndices[ idx ] )
+          if( setIndices[ idx ] != ownedIndices[ idx ] )
           {
             m_setChanged = true;
             break;
           }
         }
       }
-      m_setsIndices[ setIdx ].resize( newSetSize );
+      setIndices.resize( newSetSize );
       for( localIndex idx = 0; idx < newSetSize; ++idx )
       {
-        m_setsIndices[ setIdx ][ idx ] = ownedIndices[ idx ];
+        setIndices[ idx ] = ownedIndices[ idx ];
       }
     }
   }
 }
 
-localIndex PackCollection::getNumMetaCollectors( ) const
+localIndex PackCollection::numMetaDataCollectors( ) const
```

Comment:
couldn't `numMetaDataCollectors` be confused for a method returning a size or numeration rather than `int`/`bool` like code ?

Reply:
btw isn't it the same as `return m_targetIsMeshObject && !m_disableCoordCollection` by precedence rules ?

Reply:
Indeed, I was confused too.
It happens that the global contract seems to allow multiple `metaDataCollectors`, but the implementation allows only one (since the `bool` is 0 or 1).
I decided not to get too much in the details here and accept the pattern as it is.

Could you further explain @wrtobin because I do not have the full picture here?

> btw isn't it the same as return m_targetIsMeshObject && !m_disableCoordCollection by precedence rules ?

Indeed but I wanted to explicitly convert the `bool` in `localIndex` (this should/could be a `warning`/`error`).
I'll write a line of comment here.

Reply:
At present yes it seems there is only going to be one type of meta-collector, when we are collecting from a mesh object and want to collect coordinates for each object automatically. Previously we also added a separate 'time' collector, but that functionality got rolled into the primary collection object, so really this naming is an artifact of that combined with code intertia (in the absence of additional reasons to change it, time wasn't given to it). 

Also leaving it unspecialized for the particular case we have is a nod towards future proofing if we develop additional / specialized history collectors (other than pack), and they require different metacollection semantics (not happening anytime soon or even planned, but leaving the pattern of potentially multiple metacollectors also doesn't detract).